backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/computing/IModularContractComputingMethodHandler.java

```diff
@@ -54,24 +52,14 @@ public interface IModularContractTypeHandler<T>
 	 * </ul>
 	 */
 	@NonNull
-	Stream<FlatrateTermId> streamContractIds(@NonNull T model);
-
-	void validateAction(@NonNull final T model, @NonNull final ModelAction action);
+	Stream<FlatrateTermId> streamContractIds(@NonNull final TableRecordReference tableRecordReference);
```

Comment:
why we need this method?

Reply:
To get the FlatrateTermIds that are relevant on logCreation for different models, see `de.metas.contracts.modular.ModularContractProvider`
Also to check, if this TableReferences have valid FlatrateTerms